core/domain/feature_flag_services_test.py

```diff
@@ -0,0 +1,542 @@
+        user_3_id = self.get_user_id_from_email(user_3_email)
+
+        return (user_1_id, user_2_id, user_3_id)
+
+    def test_feature_flag_config_is_same_for_user_with_every_retrieval(
+        self) -> None:
+        user_1_id, user_2_id, user_3_id = (
+            self._signup_multiple_users_and_return_ids())
+        initial_user1_value, initial_user2_value, initial_user3_value = (
+            False, False, False)
+        with self.swap_all_feature_names_set:
+            feature_services.update_feature_flag(
+                self.dev_feature_flag.name, False, 50, [])
+            initial_user1_value = feature_services.is_feature_flag_enabled(
+                user_1_id, self.dev_feature_flag.name)
+            initial_user2_value = feature_services.is_feature_flag_enabled(
+                user_2_id, self.dev_feature_flag.name)
+            initial_user3_value = feature_services.is_feature_flag_enabled(
+                user_3_id, self.dev_feature_flag.name)
+            for _ in range(500):
+                self.assertEqual(
+                    initial_user1_value,
+                    feature_services.is_feature_flag_enabled(
+                        user_1_id,
+                        self.dev_feature_flag.name
+                    )
+                )
+                self.assertEqual(
+                    initial_user2_value,
+                    feature_services.is_feature_flag_enabled(
+                        user_2_id,
+                        self.dev_feature_flag.name
+                    )
+                )
+                self.assertEqual(
+                    initial_user3_value,
+                    feature_services.is_feature_flag_enabled(
+                        user_3_id,
+                        self.dev_feature_flag.name
+                    )
+                )
+
+    def test_feature_flag_enabled_for_users_in_5_perc_when_increased_to_10_perc(
+        self) -> None:
+        user_ids_list = []
+        user_id_to_feature_flag_status_for_5_perc = {}
+        count_feature_flag_enabled_for_5_perc = 0
+        user_id_to_feature_flag_status_for_10_perc = {}
+        count_feature_flag_enabled_for_10_perc = 0
+        for count in range(1, 1001):
+            user_ids_list.append('userid' + str(count))
+
+        with self.swap_all_feature_names_set:
+            feature_services.update_feature_flag(
+                self.dev_feature_flag.name, False, 5, [])
+            for user_id in user_ids_list:
+                feature_flag_status = feature_services.is_feature_flag_enabled(
+                    user_id,
+                    self.dev_feature_flag.name
+                )
+                if feature_flag_status:
+                    count_feature_flag_enabled_for_5_perc += 1
+                    user_id_to_feature_flag_status_for_5_perc[
+                        user_id] = feature_flag_status
+            self.assertTrue(
+                count_feature_flag_enabled_for_5_perc in list(range(40, 61)))
```

Comment:
I'm a bit confused about how I should confirm the probability for this part. Sorry about that.

Reply:
Take a binomial random variable for n = 1000 and p = 0.05. This roughly follows a normal distribution with mean np = 50 and std dev sqrt(npq) = 6.89. See e.g. https://stats.libretexts.org/Courses/Las_Positas_College/Math_40%3A_Statistics_and_Probability/06%3A_Continuous_Random_Variables_and_the_Normal_Distribution/6.04%3A_Normal_Approximation_to_the_Binomial_Distribution

In a normal distribution, around 0.997 of values are within 3 s.d. of the mean, and there is a chance of only 0.00006334 of them being outside 4 s.d. of the mean. (Reference: https://en.wikipedia.org/wiki/68%E2%80%9395%E2%80%9399.7_rule#Table_of_numerical_values) So if you don't want this test to be flaky than you'd need a range that's something like (50 - 4 * 6.89, 50 + 4 * 6.89), I think.

You can do a similar calculation for 10%. Make sure to leave an explanatory comment in both cases so that other devs know how you arrived at the relevant numbers.

Reply:
Thanls for the explanation, I have made the changes.